src/fileSyncer.ts

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import * as utilities from './utilities';
+import * as azHelper from './azureStorageHelper';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { SSHServer } from './interfaces';
+import * as sshHelper from './sshRunner';
+
+export enum TargetType {
+    cloudshell = 'cloudshell',
+    remotehost = 'remotehost'
+}
+
+export class FileSyncer {
+
+    protected _outputChannel: vscode.OutputChannel;
+
+
+    constructor(outputChannel: vscode.OutputChannel) {
+        this._outputChannel = outputChannel;
+
+    }
+
+    public syncFolderToRemoteSSHHost(): void {
+        let rootWorkspace = '';
+        if (vscode.workspace.workspaceFolders) {
+            rootWorkspace = vscode.workspace.workspaceFolders[0].uri.fsPath;
+        }
+        vscode.window.showInputBox({ value: rootWorkspace, prompt: 'Please input folder path', placeHolder: 'folder', password: false })
+            .then((folder) => {
```

Comment:
i think in VSC there's a specialized folder selector, so maybe you could try that?